src/LinkDotNet.NCronJob/Configuration/NCronJobOptionBuilder.cs

```diff
@@ -61,21 +62,86 @@ public NCronJobOptionBuilder<T> AddJob<T>(Action<JobOptionBuilder>? options = nu
         return new NCronJobOptionBuilder<T>(Services, Settings);
     }
 
-    private static CronExpression GetCronExpression(JobOption option)
+    /// <summary>
+    /// Adds a job using an asynchronous anonymous delegate to the service collection that gets executed based on the given cron expression.
+    /// </summary>
+    /// <param name="jobDelegate">The delegate that represents the job to be executed.</param>
+    /// <param name="cronExpression">The cron expression that defines when the job should be executed.</param>
+    /// <returns></returns>
+    public NCronJobOptionBuilder AddJob(Delegate jobDelegate, string cronExpression)
+    {
+        var jobName = AddJobInternal(typeof(DynamicJobFactory), jobDelegate, cronExpression);
+        Services.AddKeyedSingleton(typeof(DynamicJobFactory), jobName, (sp, _) =>
+            new DynamicJobFactory(sp, jobDelegate));
+        return this;
+    }
+    
+    /// <summary>
+    /// Registers a job in the service collection with specified delegate and cron expression. 
+    /// This method configures job options, validates the cron expression, and creates a registry entry for the job.
+    /// </summary>
+    /// <param name="jobType">The type of the job to be registered.</param>
+    /// <param name="jobDelegate">The delegate that represents the job's execution logic. This can be either a synchronous or asynchronous delegate.</param>
+    /// <param name="cronExpression">The cron expression that specifies the schedule on which the job should be executed.</param>
+    /// <returns>The name generated for the job, which is used as the key for scoped service registration.
+    /// This name is derived from the job delegate and is used to uniquely identify the job in the service collection.</returns>
+    /// <exception cref="ArgumentException">Thrown if the provided <paramref name="cronExpression"/> is null or empty.</exception>
+    private string AddJobInternal(Type jobType, Delegate jobDelegate, string cronExpression)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(cronExpression);
+
+        var determinedPrecision = JobOptionBuilder.DetermineAndValidatePrecision(cronExpression, null);
+
+        var jobOption = new JobOption
+        {
+            CronExpression = cronExpression,
+            EnableSecondPrecision = determinedPrecision,
+            TimeZoneInfo = TimeZoneInfo.Utc
+        };
+        var cron = GetCronExpression(jobOption);
+
+        var jobName = GenerateJobName(jobDelegate);
```

Comment:
While not as user-friendly - a `Guid.NewGuid()` would suffice as a name as well, doesn't it? You created a somewhat "friendly" name in regards to the dashboard?

Reply:
Exactly. This is not only required to target/lookup a specific Job but so it can be easier to read from a dashboard or CLI. There are other nicer techniques to get short unique IDs (like [squids](https://sqids.org/dotnet)) but I didn't want to rely on any external packages. This can always be changed.

Reply:
Additionally, the use of consistent job names, as opposed to generating new GUIDs, is important for several reasons:

1. **Job Identification Across Restarts**: Job names must remain constant across process restarts to ensure accurate mapping and tracking of job run history.
2. **Distributed Environments**: Consistent job names are essential in scenarios involving multiple instances of the same process. This consistency aids in managing distributed locking mechanisms effectively.

Because we're not deriving from IJob we can't get the Job type's Name because one doesn't exist. This approach generates the a name from the delegate's signature, so as long as the signature doesn't change the name will be consistent. Using a stable identifier for each job enhances our ability to maintain state and coordination across system restarts and distributed deployments, improving reliability and traceability.

Reply:
I do like that - don’t get me wrong I wasn’t pushing towards Guid.NewGuid - more interested in the reasoning!